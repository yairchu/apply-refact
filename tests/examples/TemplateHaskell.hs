{-# LANGUAGE TemplateHaskellQuotes #-}

import Language.Haskell.TH

t = varE 'id
