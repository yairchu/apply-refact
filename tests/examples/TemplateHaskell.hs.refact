[ ( "tests/examples/TemplateHaskell.hs:5:5-12: Suggestion: Suggestion: Redundant varE\nFound:\n  varE 'id\nPerhaps:\n  [| id |]\nNote: may require `{-# LANGUAGE TemplateHaskell #-}` adding to the top of the file\n"
  , [ Replace {rtype = Pattern, pos = SrcSpan {startLine = 5, startCol = 5, endLine = 5, endCol = 13}, subts = [("[| id |]",SrcSpan {startLine = 5, startCol = 5, endLine = 5, endCol = 13})], orig = "varE 'id"}
    , InsertComment {pos = SrcSpan {startLine = 1, startCol = 1, endLine = 1, endCol = 34}, newComment = "{-# LANGUAGE TemplateHaskell #-}"}
    ]
  )
]
