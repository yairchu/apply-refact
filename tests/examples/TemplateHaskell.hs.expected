{-# LANGUAGE TemplateHaskell #-}
{-# LANGUAGE TemplateHaskellQuotes #-}

import Language.Haskell.TH

t = [|id|]
